src/resolve/resolved-dns-scope.c

```diff
@@ -1365,7 +1365,18 @@ bool dns_scope_name_wants_search_domain(DnsScope *s, const char *name) {
         if (s->protocol != DNS_PROTOCOL_DNS)
                 return false;
 
-        return dns_name_is_single_label(name);
+        bool is_single_label = dns_name_is_single_label(name);
```

Comment:
just do 

```c
if (!dns_name_is_single_label(name))
       return false;
```

here, and you don't need to have the boolean or check anything anymore.